component.js

```diff
@@ -255,9 +259,21 @@ function factory (options) {
     }
 
     if (!hasShouldComponentUpdate(mixins)) {
-      mixins = [{
+      mixins.unshift({
         shouldComponentUpdate: _shouldComponentUpdate
-      }].concat(mixins);
+      });
+    }
+
+    if (!hasComponentWillReceiveProps(mixins)) {
```

Comment:
@torgeir @mikaelbr I can do that, but unless I'm mistaken passed mixins can not have a conflicting methods, which means that omniscient would have to detect if any mixin has one of these methods and wrap over it.

There reason I did not do it was to avoid changes to current behavior, I don't think omniscient deals with conflicts in mixins, or am I mistaken ?